lib/std/http.zig

```diff
@@ -784,21 +776,7 @@ pub const BodyWriter = struct {
     pub fn flush(w: *BodyWriter) Error!void {
         const out = w.http_protocol_output;
         switch (w.state) {
-            .end, .none, .content_length => return out.flush(),
-            .chunked => |*chunked| switch (chunked.*) {
-                .offset => |offset| {
-                    const chunk_len = out.end - offset - chunk_header_template.len;
-                    if (chunk_len > 0) {
-                        writeHex(out.buffer[offset..][0..chunk_len_digits], chunk_len);
-                        chunked.* = .{ .chunk_len = 2 };
-                    } else {
-                        out.end = offset;
-                        chunked.* = .{ .chunk_len = 0 };
-                    }
-                    try out.flush();
-                },
-                .chunk_len => return out.flush(),
-            },
```

Comment:
to give you some background for why this was done: the idea was that when flushing a chunked http transfer request, it should finish the chunk and send the end-of-chunk trailer and start a new chunk. Looking at it with fresh eyes, I'm not sure how valuable that is, although I can think of some reasons for it.

anyway I think the simplification that you brought is nice, so let's run with that.

Reply:
Hmm, this offset code in flush was actually unreachable before my commit.

I don't think it's possible to implement what you describe either. We can't finish the chunk early without writing the full amount of data specified in the chunk header, and we don't have access to data that the user of the API started but did not finish writing here in flush().